_plugins/redirect_json.rb

```diff
@@ -0,0 +1,43 @@
+# Creates redirect pages given a redirects.json file.
+# Redirects.json should have the format:
+# {
+#   "/source-url": "https://destination-url/",
+#   ...
+# }
+
+module RedirectsJson
+  class RedirectsJsonPage < Jekyll::Page
+    # Use Jekyll's native absolute_url filter
+    include Jekyll::Filters::URLFilters
+
+    def initialize(site, from, to)
+      from += '.html' unless from.end_with?('.html')
+
+      @site = site
+      @name = from
```

Comment:
Why are you creating an instance variable from the input from? It also looks like @name is never referenced after being processed?

Reply:
There are things that Jekyll + plugins make use of internally to get information about pages. [Here's the source](https://github.com/jekyll/jekyll/blob/master/lib/jekyll/page.rb#L33-L36), and [here's how jekyll-redirect does it](https://github.com/jekyll/jekyll-redirect-from/blob/master/lib/jekyll-redirect-from/redirect_page.rb#L5)

Calling the builtin `initialize` for a page does _too much stuff_ (reading yaml, etc) so I set the necessary metadata here to avoid adding noops for all of those other methods.

Reply:
Ok yeah. Sorry I'm just confused because I'm not seeing the @name variable referenced here after being processed, like in the Jekyll docs? If the intent is to process the from url, process() mutates the data in place, so the processed value is not being used here.

Reply:
All clear!